sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -206,6 +206,18 @@ trait CodegenSupport extends SparkPlan {
   def doConsume(ctx: CodegenContext, input: Seq[ExprCode], row: ExprCode): String = {
     throw new UnsupportedOperationException
   }
+
+  /**
```

Comment:
Suggestion for the comment:

    /**
     * For optimization to suppress `shouldStop()` in a loop of WholeStageCodegen.
     * Returning true means we need to insert `shouldStop()` into the loop producing rows, if any.
     */